src/TestFramework/Codeception/Adapter/CodeceptionAdapter.php

```diff
@@ -0,0 +1,395 @@
+        $this->stringifier = $stringifier;
+        $this->srcDirs = $srcDirs;
+    }
+
+    public function hasJUnitReport(): bool
+    {
+        return true;
+    }
+
+    public function testsPass(string $output): bool
+    {
+        if (preg_match('/failures!/i', $output)) {
+            return false;
+        }
+
+        if (preg_match('/errors!/i', $output)) {
+            return false;
+        }
+
+        // OK (XX tests, YY assertions)
+        $isOk = preg_match('/OK\s\(/', $output);
+
+        // "OK, but incomplete, skipped, or risky tests!"
+        $isOkWithInfo = preg_match('/OK\s?,/', $output);
+
+        // "Warnings!" - e.g. when deprecated functions are used, but tests pass
+        $isWarning = preg_match('/warnings!/i', $output);
+
+        return $isOk || $isOkWithInfo || $isWarning;
+    }
+
+    public function getMemoryUsed(string $output): float
+    {
+        if (preg_match('/Memory: (\d+(?:\.\d+))\s*MB/', $output, $match)) {
+            return (float) $match[1];
+        }
+
+        return -1;
+    }
+
+    public function getName(): string
+    {
+        return TestFrameworkTypes::CODECEPTION;
+    }
+
+    public function getInitialTestRunCommandLine(string $extraOptions, array $phpExtraArgs, bool $skipCoverage): array
+    {
+        $argumentsAndOptions = $this->prepareArgumentsAndOptions($extraOptions);
+
+        return $this->commandLineBuilder->build(
+            $this->testFrameworkExecutable,
+            $phpExtraArgs,
+            array_merge(
+                $argumentsAndOptions,
+                [
+                    '--coverage-phpunit',
+                    XMLLineCodeCoverage::CODECEPTION_COVERAGE_DIR,
+                    // JUnit report
+                    '--xml',
+                    $this->jUnitFilePath,
+                    '-o',
+                    "paths: output: {$this->tmpDir}",
+                    '-o',
+                    sprintf('coverage: enabled: %s', $this->stringifier->stringifyBoolean(!$skipCoverage)),
+                    '-o',
+                    sprintf('coverage: include: %s', $this->getCoverageIncludeFiles($skipCoverage)),
+                    '-o',
+                    'settings: shuffle: true',
+                ]
+            )
+        );
+    }
+
+    public function getMutantCommandLine(MutantInterface $mutant, string $extraOptions): array
+    {
+        $argumentsAndOptions = $this->prepareArgumentsAndOptions($extraOptions);
+
+        $commandLine = $this->commandLineBuilder->build($this->testFrameworkExecutable, [], $argumentsAndOptions);
+
+        $output = sprintf('%s/%s', $this->tmpDir, $mutant->getMutation()->getHash());
+
+        $interceptorFilePath = sprintf(
+            '%s/interceptor.codeception.%s.php',
+            $this->tmpDir,
+            $mutant->getMutation()->getHash()
+        );
+
+        file_put_contents($interceptorFilePath, $this->createCustomBootstrapWithInterceptor($mutant), LOCK_EX);
```

Comment:
this should rely on the `Safe\*` version

Reply:
done